src/API/Google/AdsCampaign.php

```diff
@@ -382,32 +382,27 @@ public function delete_campaign( int $campaign_id ): int {
 	 * @return array
 	 */
 	public function get_highest_spend_campaign(): array {
-		$highest_spend_campaign = [];
-
 		try {
 			$campaigns = $this->get_campaigns();
 		} catch ( Exception $e ) {
-			return $highest_spend_campaign;
+			return [];
 		}
 
-		foreach ( $campaigns as $campaign ) {
-			// Skip campaign if not enabled.
-			if ( CampaignStatus::ENABLED !== $campaign['status'] ) {
-				continue;
-			}
-
-			// Set the first campaign found as the highest campaign.
-			if ( empty( $highest_spend_campaign ) ) {
-				$highest_spend_campaign = $campaign;
-				continue;
-			}
+		return array_reduce(
+			$campaigns,
+			function ( $highest, $campaign ) {
+				if ( empty( $highest ) ) {
+					return $campaign;
+				}
```

Comment:
One thing I just noticed while reviewing this change is that this could result in returning a non-enabled campaign. I think a slight refactor would avoid this problem:

```php
function ( $highest, $campaign ) {
	if ( CampaignStatus::ENABLED === $campaign['status'] && (
		empty( $highest ) || $campaign['amount'] > $highest['amount'] )
		) {
		return $campaign;
	}

	return $highest;
}
```

Reply:
Thanks @joemcgill I've updated this with your suggested change